packages/discovery-react-components/src/components/StructuredQuery/components/RuleRow/RuleRow.tsx

```diff
@@ -1,44 +1,81 @@
 import React, { FC } from 'react';
 import { ComboBox, TextInput } from 'carbon-components-react';
+import { Button } from 'carbon-components-react';
+import SubtractAlt16 from '@carbon/icons-react/lib/subtract--alt/16';
 import { Messages } from 'components/StructuredQuery/messages';
 import { structuredQueryRulesClass } from 'components/StructuredQuery/cssClasses';
+import {
+  Row,
+  StructuredQuerySelection
+} from 'components/StructuredQuery/utils/structuredQueryInterfaces';
 
 export interface RuleRowProps {
   /**
    * override default messages for the component by specifying custom and/or internationalized text strings
    */
   messages: Messages;
+  /**
+   * id of the rule row to render
+   */
+  rowId: Row['id'];
+  /**
+   * state that represents the current rules and selections for the structured query
+   */
+  ruleRows: StructuredQuerySelection;
+  /**
+   * used to set the ruleRows state
+   */
+  setRuleRows: (ruleRows: StructuredQuerySelection) => void;
 }
 
-export const RuleRow: FC<RuleRowProps> = ({ messages }) => {
+export const RuleRow: FC<RuleRowProps> = ({ messages, rowId, ruleRows, setRuleRows }) => {
   const operatorDropdownItems = [
     { label: messages.operatorDropdownIsOptionText, value: '::' },
     { label: messages.operatorDropdownIsNotOptionText, value: '::!' },
     { label: messages.operatorDropdownContainsOptionText, value: ':' },
     { label: messages.operatorDropdownDoesNotContainOptionText, value: ':!' }
   ];
+  const showRemoveRuleRowButton = ruleRows.rows.length > 1;
+
+  const handleRemoveRuleRowOnClick = () => {
+    setRuleRows(
+      Object.assign({}, ruleRows, {
+        rows: ruleRows.rows.filter(ruleRow => ruleRow.id !== rowId)
```

Comment:
is the plan for the `ruleRows` object to have more attributes? or could this just be an array that we do the same operation on?

also `Object.assign` can usually be replaced with `{...ruleRows, rows: ruleRows.rows.filter()}`

Reply:
Yeah, my plan is that it will have more attributes. And will replace 👍